src/js_native_api_v8.cc

```diff
@@ -314,16 +314,29 @@ class RefBase : protected Finalizer, RefTracker {
 };
 
 class Reference : public RefBase {
+  class PersistentWeakParameter {
+   public:
+    explicit PersistentWeakParameter(Reference* reference)
+        : _reference(reference) {}
+
+    inline Reference* reference() { return _reference; }
+
+    inline void Reset(Reference* reference = nullptr) {
+      _reference = reference;
+    }
+
+   private:
+    Reference* _reference;
+  };
```

Comment:
The lifetime of this new class is detached from the `Reference` once there are second pass callbacks scheduled. So that the `Reference` can be released in regards to `node_env__` teardown, and the second pass callback can still be safely invoked by the platform task runner in delayed.

Reply:
Edited the OP to link back to the original discussion: https://github.com/nodejs/node/pull/37802#discussion_r604722450

Reply:
Okay, but … I didn’t talk about lifetimes? My point is that you could essentially also just do `using PersistentWeakParameter = Reference*;` here, and get essentially the same result

Reply:
The class here essentially holds the `Reference*` and we can reset the value once the second pass callback is scheduled. If the parameter value in `WeakCallbackInfo` could be changed during callback, then it is not necessary to wrap the pointers... Ooops, let me check if that is possible.

Reply:
Oh, I see your suggestion here. Yeah, totally make sense...